pkg/utilities/manifest/manifest.go

```diff
@@ -1,248 +1,144 @@
 package manifest
 
 import (
-	encjson "encoding/json"
-	"io/ioutil"
+	"bytes"
+	"io"
 	"os"
-	"reflect"
 	"strings"
 
-	"github.com/pkg/errors"
-	bmv1alpha3 "github.com/weaveworks/wksctl/pkg/existinginfra/v1alpha3"
-	"gopkg.in/oleiade/reflections.v1"
-	"k8s.io/apimachinery/pkg/api/meta"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/runtime/schema"
-	"k8s.io/apimachinery/pkg/runtime/serializer/json"
-	"k8s.io/client-go/kubernetes/scheme"
-	clusterv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
-	"sigs.k8s.io/yaml"
+	"github.com/weaveworks/libgitops/pkg/serializer"
+	kyaml "sigs.k8s.io/kustomize/kyaml/yaml"
 )
 
-// cluster-api types
 const (
-	GroupName        = "sigs.k8s.io"
-	GroupVersion     = "v1alpha3"
 	DefaultNamespace = `weavek8sops`
+
+	corev1Version = "v1"
+	listKind      = "List"
+	namespaceKind = "Namespace"
 )
 
 var DefaultAddonNamespaces = map[string]string{"weave-net": "kube-system"}
 
-var SchemeGroupVersion = schema.GroupVersion{Group: GroupName, Version: GroupVersion}
-var SchemeBuilder = runtime.NewSchemeBuilder(addKnownTypes)
-var AddToScheme = SchemeBuilder.AddToScheme
+func WithNamespace(fileOrString, namespace string) ([]byte, error) {
```

Comment:
Not changed in this PR, but `fileOrString` seems weird : it's only used as a filename in one place, but in several other places what is passed in is the result from `ioutil.ReadAll()`.  Anyway it could easily be two functions.

Reply:
Changed to use `io.ReadCloser` in faf7480, this removes redundant conversions and enables quite smooth usage with both files and bytes.